File: app/src/main/java/de/tum/in/tumcampusapp/models/barrierfree/BarrierfreeMoreInfo.java
package de.tum.in.tumcampusapp.models.barrierfree;

import de.tum.in.tumcampusapp.adapters.SimpleStickyListHeadersAdapter;

/**
 * The model used to display more infromation in barrier free page
 */
public class BarrierfreeMoreInfo implements SimpleStickyListHeadersAdapter.SimpleStickyListItem{
//    private int id;
    private String title;
    private String category;
    private String url;

    public BarrierfreeMoreInfo(String title, String category, String url) {
        this.title = title;
        this.category = category;
        this.url = url;
    }

    public String getTitle() {
        return title;
    }

    public String getCategory() {
        return category;
    }

    public String getUrl() {
        return url;
    }

    @Override
    public String getHeadName() {
        return getCategory();
    }
}

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/barrierfree/model/BarrierFreeMoreInfo.kt
package de.tum.`in`.tumcampusapp.component.ui.barrierfree.model

import de.tum.`in`.tumcampusapp.component.other.generic.adapter.SimpleStickyListHeadersAdapter

/**
 * The model used to display more infromation in barrier free page
 */
data class BarrierFreeMoreInfo(
    var title: String = "",
    var category: String = "",
    var url: String = ""
) : SimpleStickyListHeadersAdapter.SimpleStickyListItem {

    override fun getHeadName() = category

    override fun getHeaderId() = category
}
